### PR TITLE
help ; now has output

### PR DESCRIPTION
**Pull request by willywalters5**
Fixes issue #34842.
The semicolon output suppression prevented the output from being displayed.

### CODE REVIEW

File: stdlib/REPL/src/REPL.jl [hunk 1]
```diff
@@ -743,7 +743,7 @@ function eval_with_backend(ast, backend::REPLBackendRef)
     take!(backend.response_channel) # (val, iserr)
 end
 
-function respond(f, repl, main; pass_empty = false)
+function respond(f, repl, main; pass_empty = false, supress_on_semicolon = true)
```
Comment (JeffBezanson) [inline stdlib/REPL/src/REPL.jl]:
```suggestion
function respond(f, repl, main; pass_empty = false, suppress_on_semicolon = true)
```

File: stdlib/REPL/src/REPL.jl [hunk 2]
```diff
@@ -758,7 +758,8 @@ function respond(f, repl, main; pass_empty = false)
             catch
                 response = (catch_stack(), true)
             end
-            print_response(repl, response, !ends_with_semicolon(line), Base.have_color)
+	    hide_output = supress_on_semicolon && ends_with_semicolon(line)
```
Comment (JeffBezanson) [inline stdlib/REPL/src/REPL.jl]: Indentation needs fixing here; use spaces instead of tab.

**Comment by JeffBezanson** [comment]
Thanks!

**Review by fredrikekre — APPROVED** [review]

**Comment by fredrikekre** [comment]
This makes it impossible to suppress help output though, e.g. this would not "work":
```
help?> sin;

julia>
```
but I guess it's kind of pointless to request suppressed help.

**Pull request merged**